File: JS, PHP, PYTHON/PHP/INTERACCION.php
<?php
function adivinaElNumero() {
    $num = rand(1, 10); // Genera un número aleatorio entre 1 y 10
    $adivinar = (int)readline("Adivina el número del 1 al 10: ");

    if ($num === $adivinar) {
        echo "Has ganado\n";
    } else {
        echo "Has perdido, el número era: $num\n";
    }
}

adivinaElNumero();
?>

File: C++ y Java/C++/Interaccion.cpp
#include <iostream>
#include <cstdlib>
#include <ctime>

int main()
{

  std::srand(std::time(0));

  int numeroAleatorio = std::rand() % 10 + 1;

  int numeroUsuario;
  std::cout << "Adivina el numero del 1 al 10: ";
  std::cin >> numeroUsuario;

  if (numeroUsuario == numeroAleatorio)
  {
    std::cout << "Has ganado" << std::endl;
  }
  else
  {
    std::cout << "Has perdido, el numero era: " << numeroAleatorio << std::endl;
  }

  return 0;
}